packages/core/strapi/lib/services/server/register-routes.js

```diff
@@ -31,17 +31,29 @@ module.exports = (strapi) => {
  * @param {import('../../').Strapi} strapi
  */
 const registerAdminRoutes = (strapi) => {
-  const generateRouteScope = createRouteScopeGenerator(`admin::`);
+  const adminRoutes = strapi.admin.routes;
+  const generateAdminRouteScope = createRouteScopeGenerator(`admin::`);
 
-  _.forEach(strapi.admin.routes, (router) => {
+  const buildRoutesFromRouter = (router, apiName = '') => {
+    if (Array.isArray(router)) {
+      return _.forEach(router, (routerItem) => buildRoutesFromRouter(routerItem, apiName));
+    }
```

Comment:
will it not always be an array by how it is now defined?

Reply:
No, you have other routes that are objects. 👍